label_studio/core/middleware.py

```diff
@@ -159,6 +159,21 @@ def __call__(self, request):
         return response
 
 
+class XApiKeySupportMiddleware:
+    """Middleware that adds support for the X-Api-Key header, by having its value supersede
+    anything that's set in the Authorization header."""
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if 'HTTP_X_API_KEY' in request.META:
+            request.META['HTTP_AUTHORIZATION'] = f'Token {request.META["HTTP_X_API_KEY"]}'
+            del request.META['HTTP_X_API_KEY']
```

Comment:
why should we remove it from the payload?

Reply:
It isn't used otherwise, and I just wanted to make sure that we won't have a situation where different auth headers are read in different spots on the same request. This is also why this middleware is set to run right before AuthenticationMiddleware